lib/components/popups/course-search/query-components/Form.tsx

```diff
@@ -133,11 +97,13 @@ const Form: FC<{ setSearching: (searching: boolean) => void }> = (props) => {
       );
   };
 
-  // Search with debouncing of 2/4s of a second.
-  const minLength = 4;
   useEffect(() => {
-    searchedCourses.clear();
-    setInitialQueryLength(searchTerm.length);
+    setPageIndex(0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm, searchFilters]);
+
+  // Search with debouncing of 4/8s of a second.
+  useEffect(() => {
```

Comment:
useEffect that calls /search route and updates retrievedCourses every time [searchTerm, searchFilters, pageIndex] change